### PR TITLE
[Feature] Register torchvision transforms into mmcls

### PR DESCRIPTION
**Pull request by Ezra-Yu**
## Motivation

Register torchvision transforms into mmcls

## Modification

0. Add stochastic depth decay rule in resnet refer to https://github.com/open-mmlab/mmpretrain/pull/1363
1. Register torchvision transforms into mmcls;
2. update transform `NumpyToPIL` and `PILToNumpy`;

### 1. Use some vision Augs Surrounded by `NumpyToPIL` and `'PILToNumpy`  (Recommendation)

Surrounded by ``dict(type='NumpyToPIL', to_rgb=True),`` and ``dict(type='PILToNumpy', to_bgr=True),``

```
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='NumpyToPIL', to_rgb=True),     # from BGR in cv2 to RGB  in PIL
    dict(type='torchvision/RandomResizedCrop',size=176),
    dict(type='PILToNumpy', to_bgr=True),     # from RGB  in PIL to BGR in cv2
    dict(type='RandomFlip', prob=0.5, direction='horizontal'),
    dict(type='PackInputs'),
]

data_preprocessor = dict(
    num_classes=1000,
    mean=[123.675, 116.28, 103.53],
    std=[58.395, 57.12, 57.375],
    to_rgb=True,                          # from BGR in cv2 to RGB  in PIL
)
```

### 2. Use vision Augs and ToTensor&Normalize
Make sure have coverted to RGB-Numpy format; `` dict(type='NumpyToPIL', to_rgb=True)``
```
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='NumpyToPIL', to_rgb=True),       # from BGR in cv2 to RGB  in PIL
    dict(
        type='torchvision/RandomResizedCrop',
        size=176,
        interpolation=2),
    dict(type='torchvision/RandomHorizontalFlip', p=0.5),
    dict(
        type='torchvision/TrivialAugmentWide',
        interpolation=2),
    dict(type='torchvision/PILToTensor'),
    dict(type='torchvision/ConvertImageDtype', dtype=torch.float),
    dict(      
        type='torchvision/Normalize',  
        mean=(0.485, 0.456, 0.406),
        std=(0.229, 0.224, 0.225),
    ),
    dict(type='torchvision/RandomErasing', p=0.1),
    dict(type='PackInputs'),
]

data_preprocessor = dict(num_classes=1000, mean=None, std=None, to_rgb=False)  # Normalize in dataset pipeline
```
### 3. USe Augs but no ToTensor&Normalize
```
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='NumpyToPIL', to_rgb=True),   # from BGR in cv2 to RGB  in PIL
    dict(type='torchvision/RandomResizedCrop', size=176, interpolation=2),
    dict(type='torchvision/RandomHorizontalFlip', p=0.5),
    dict(type='torchvision/TrivialAugmentWide', interpolation=2),
    dict(type='PackInputs'),
]

# here the Normalize params is for the RGB fromat
data_preprocessor = dict(
    num_classes=1000,
    mean=[123.675, 116.28, 103.53],
    std=[58.395, 57.12, 57.375],
    to_rgb=False,
)
```

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcls/datasets/transforms/formatting.py [hunk 1]
```diff
@@ -141,55 +142,70 @@ def __repr__(self):
...
+        results['img'] = Image.fromarray(img)
         return results
 
+    def __repr__(self):
+        return self.__class__.__name__ + f'(to_rgb={self.to_rgb})'
 
-@TRANSFORMS.register_module()
+
+@TRANSFORMS.register_module(('ImgToNumpy', 'ToNumpy'))
 class ToNumpy(BaseTransform):
```
Comment (YuanLiuuuuuu) [inline mmcls/datasets/transforms/formatting.py:177]: Maybe PILToNumpy is better?

File: mmcls/datasets/transforms/formatting.py [hunk 2]
```diff
@@ -141,55 +142,70 @@ def __repr__(self):
             f'(keys={self.keys}, order={self.order})'
 
 
-@TRANSFORMS.register_module()
+@TRANSFORMS.register_module(('ImgToPIL', 'ToPIL'))
 class ToPIL(BaseTransform):
```
Comment (YuanLiuuuuuu) [inline mmcls/datasets/transforms/formatting.py:146]: NumpyToPIL?

File: mmcls/datasets/transforms/formatting.py [hunk 3]
```diff
@@ -141,55 +142,70 @@ def __repr__(self):
...
     **Modified Keys:**
 
-    - img
+    - ``*img**``
+
+    Args:
+        to_rgb (bool): Whether to convert img to rgb. Defaults to False.
     """
 
+    def __init__(self, to_rgb: bool = False):
```
Comment (YuanLiuuuuuu) [inline mmcls/datasets/transforms/formatting.py:161]:
```suggestion
    def __init__(self, to_rgb: bool = False) -> None:
```

Comment (YuanLiuuuuuu) [reply mmcls/datasets/transforms/formatting.py:161]: Please also check other return typehint

File: mmcls/datasets/transforms/formatting.py [hunk 4]
```diff
@@ -213,55 +215,70 @@ def __repr__(self):
...
 
     Args:
+        to_rgb (bool): Whether to convert img to rgb. Defaults to False.
         dtype (str, optional): The dtype of the converted numpy array.
             Defaults to None.
     """
 
-    def __init__(self, keys, dtype=None):
-        self.keys = keys
+    def __init__(self, to_rgb: bool = False, dtype=None) -> None:
```
Comment (YuanLiuuuuuu) [inline mmcls/datasets/transforms/formatting.py:267]: should be to_bgr

File: mmcls/datasets/transforms/formatting.py [hunk 5]
```diff
@@ -213,55 +215,70 @@ def __repr__(self):
...
+
+    Args:
+        to_rgb (bool): Whether to convert img to rgb. Defaults to False.
     """
 
-    def transform(self, results):
+    def __init__(self, to_rgb: bool = False) -> None:
+        self.to_rgb = to_rgb
+
+    def transform(self, results: Dict) -> Dict:
```
Comment (YuanLiuuuuuu) [inline mmcls/datasets/transforms/formatting.py:237]:
```suggestion
    def transform(self, results: dict) -> dict:
```
Please also check other parts

File: docs/en/api/data_process.rst [hunk 6]
```diff
@@ -147,6 +147,88 @@ Transform Wrapper
...
+            std=(0.229, 0.224, 0.225),
+        ),
+        dict(type='torchvision/RandomErasing', p=0.1),
+        dict(type='PackInputs'),
+    ]
+
+    data_preprocessor = dict(num_classes=1000, mean=None, std=None, to_rgb=False)  # Normalize in dataset pipeline
+
+
+**3. USe TorchVision Augs Except ToTensor&Normalize**
```
Comment (fangyixiao18) [inline docs/en/api/data_process.rst:210]:
```suggestion
**3. Use TorchVision Augs Except ToTensor&Normalize**
```

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Review by fangyixiao18 — APPROVED** [review]

**Pull request merged**